commit 259b43dfa9cd
Author: JW Wang <jwwang@mozilla.com>
Date:   Wed Oct 18 16:08:33 2017 +0800

    Bug 1409649 - Shut down the decoder when MEDIA_ERR_ABORTED is received. r=gerald, a=ritu
    
    http://searchfox.org/mozilla-central/rev/dca019c94bf3a840ed7ff50261483410cfece24f/dom/html/HTMLMediaElement.cpp#1395-1400
    
    We need to shut down the decoder after setting network state to EMPTY.
    Otherwise the decoder will keep loading and change ready state to HAVE_METADATA
    or greater and then fail the assertion in AssertReadyStateIsNothing().
    
    MozReview-Commit-ID: FpMDVAJHTS5
    
    --HG--
    extra : source : 6a815fb0b667075041404ebd0b1ee7b87f46e3a2
---
 dom/html/HTMLMediaElement.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git dom/html/HTMLMediaElement.cpp dom/html/HTMLMediaElement.cpp
index d1f12e7bceed..5d51851047cf 100644
--- dom/html/HTMLMediaElement.cpp
+++ dom/html/HTMLMediaElement.cpp
@@ -1376,6 +1376,9 @@ public:
         mOwner->DispatchAsyncEvent(NS_LITERAL_STRING("abort"));
         mOwner->ChangeNetworkState(nsIDOMHTMLMediaElement::NETWORK_EMPTY);
         mOwner->DispatchAsyncEvent(NS_LITERAL_STRING("emptied"));
+        if (mOwner->mDecoder) {
+          mOwner->ShutdownDecoder();
+        }
       } else if (aErrorCode == MEDIA_ERR_SRC_NOT_SUPPORTED) {
         mOwner->ChangeNetworkState(nsIDOMHTMLMediaElement::NETWORK_NO_SOURCE);
       } else {
